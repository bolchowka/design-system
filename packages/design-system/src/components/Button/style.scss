@import '~@livechat/design-system-themes/dist/scss';

$base-class: "btn";

@mixin shared() {
  border-width: 1px;
  border-style: solid;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
  font-weight: 600;
  display: inline-flex;
  line-height: 1.5;
  position: relative;
  padding: 6px 16px;
  box-sizing: border-box;
  justify-content: center;
  align-items: center;
  font-size: 15px;
  min-width: 36px;
  min-height: 36px;
  transition-property: opacity, border, color, background-color, box-shadow;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0.64, 0, 0.35, 1);
  user-select: none;
  text-decoration: none;

  &:focus,
  &:hover,
  &:active {
    outline: 0;
  }

  &:disabled {
    cursor: not-allowed;
  }

  &--disabled {
    opacity: 0.65;
  }

  &--full-width {
    width: 100%;
  }

  &--large {
    font-size: 15px;
    padding: 9px 24px;
    min-width: 42px;
    min-height: 42px;
  }

  &--compact {
    font-size: 15px;
    padding: 4px 16px;
    min-width: 32px;
    min-height: 32px;
  }

  &--loading {
    transition: all 0.2s cubic-bezier(0.64, 0, 0.35, 1);

    > * {
      opacity: 0;
    } 
  }

  &__loader {
    position: absolute;
    top: 0;
    left: 0;
    opacity: 1;
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    padding: inherit;
  }

  &__loader-label {
    text-align: left;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: inherit;
    margin: 0 5px;
  }

  &__icon {
    height: 20px;
    width: 20px;
    display: inline-block;
    line-height: 1;
    margin-right: 5px;

    svg {
      box-sizing: border-box;
      height: inherit;
      width: inherit;
      fill: currentColor;
      font-style: normal;
      line-height: 0;
      text-align: center;
      text-transform: none;
      text-rendering: optimizeLegibility;
      -webkit-font-smoothing: antialiased;
      display: inline-block;
    }
  }
}

@mixin themable() {
  background-color: $basic;
  color: $primary;
  border-color: $primary;

  &:hover {
    background-color: $basic-hover;
  }

  &:focus {
    box-shadow: 0 0 1px 2px rgba($primary, 0.7);
  }

  &:active {
    box-shadow: 0 0 0 1px $border;
    transition-duration: 100ms;
  }

  &--primary {
    background-color: $primary;
    color: $ui-background;
    border-color: $primary;

    &:hover {
      background-color: $primary-hover;
    }

    &:active {
      box-shadow: 0 0 0 1px $primary;
    }
  }

  &--secondary {
    color: $text-primary;
    border-color: $border;
  }

  &--destructive {
    background-color: $danger;
    color: $ui-background;
    border-color: $danger;

    &:hover {
      background-color: $danger-hover;
    }

    &:active {
      box-shadow: 0 0 0 1px $danger;
    }
  }

  &--loading {
    background-color: $basic-disabled;
    color: transparent;

    &.btn--secondary {
      border-color: $border;
    }

    &.btn--destructive {
      background-color: $danger-disabled;
    }

    &.btn--primary {
      background-color: $primary-disabled;
    }

    &.btn--destructive,
    &.btn--primary {
      border-color: transparent;
    }
  }
}

.#{$base-class} {
  @include shared();

  @include lcds-theme() {
    @include themable();
  }
}

.theme--light {
  .#{$base-class} {
    @include lcds-theme($lcds-theme--light) {
      @include themable();
    }
  }
}
