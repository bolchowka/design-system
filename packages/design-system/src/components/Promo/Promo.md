### Promo

## Usage

Promo banner component is used to let users know about upgrade options, new features, or any other information that doesn't require immediate action. This component should be used wisely so don't overuse it too often. Remember to always specify the action that you want to promote.

### Small

This is appropriate for components from 180px to 400px width. Use it in sidebars, mobile and any other contextual use, where space is the limitation.

```js
<Promo size="small" header="This example headline has 40 characters" onClose={()=>window.alert('close')} linkText="Link button" buttonText="Primary action"   img="https://via.placeholder.com/100"
> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js
<Promo size="small" header="This example headline has 40 characters" onClose={()=>window.alert('close')} linkText="Link button" buttonText="Primary action"
> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js noeditor
<ComponentHtmlMarkup>
<Promo size="small" header="This example headline has 40 characters" onClose={()=>window.alert('close')}> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
</ComponentHtmlMarkup>
```

### Medium

This is appropriate for components from 400px to around 800px width.

Max text colum width is 520px

```js
<Promo size="medium" linkText="Link button" buttonText="Primary action" header="This example headline has 40 characters" onClose={()=>window.alert('close')} img="https://via.placeholder.com/100"> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js
<Promo size="medium" linkText="Link button" buttonText="Primary action" header="This example headline has 40 characters" onClose={()=>window.alert('close')} > A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js noeditor
<ComponentHtmlMarkup>
<Promo size="medium" header="This example headline has 40 characters" onClose={()=>window.alert('close')}> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
</ComponentHtmlMarkup>
```

### Large

Max text colum width is 520px

```js
<Promo size="large" header="This example headline has 40 characters"  linkText="Link button" buttonText="Primary action" onClose={()=>window.alert('close')} img="https://via.placeholder.com/100"> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js
<Promo size="large" header="This example headline has 40 characters"  linkText="Link button" buttonText="Primary action" onClose={()=>window.alert('close')}> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
```

```js noeditor
<ComponentHtmlMarkup>
<Promo size="large" header="This example headline has 40 characters" onClose={()=>window.alert('close')}> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
</ComponentHtmlMarkup>
```

### Background

#### On white
```js
<div style={{ padding: '20px', backgroundColor: '#fff' }}>
<Promo size="small" header="This example headline has 40 characters" onClose={()=>window.alert('close')} linkText="Link button" buttonText="Primary action"> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
</div>
```
#### On greys
```js
<div style={{ padding: '20px', backgroundColor: '#f3f7f9', width: 'fit-content' }}>
<Promo light size="small" header="This example headline has 40 characters" onClose={()=>window.alert('close')} linkText="Link button" buttonText="Primary action"> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Promo>
</div>
```