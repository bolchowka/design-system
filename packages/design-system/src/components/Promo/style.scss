@import '../../scss/variables/colors';

$base-class: 'promo';

.#{$base-class} {
  position: relative;
  padding: 20px;
  word-break: break-word;
  width: 100%;
  box-sizing: border-box;
  background-color: #f3f7f9;
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-width: 180px;

  &--light {
    border: solid 1px #dde2e6;
    background-color: #ffffff;
  }

  &--small {
    max-width: 400px;

    flex-wrap: wrap;

    .#{$base-class}__content {
      margin-bottom: 12px;
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      flex-direction: column;
    }
    .#{$base-class}__img {
      width: 48px;
      height:48px;
      margin-bottom: 12px;
    }
    .#{$base-class}__wrapper {
      display: flex;
      flex-direction: column;
    }
    .#{$base-class}__header {
      font-weight: 600;
    }

    .#{$base-class}__footer {
      display: flex;
      margin-top: 12px;
    }

    .#{$base-class}__button-text {
      margin-right: 24px;
    }
  }

  &--medium {
    max-width: 800px;
    flex-wrap: wrap;

    .#{$base-class}__content {
      margin-bottom: 12px;
      display: flex;
      justify-content: flex-start;
      flex-direction: row;
    }
    
    .#{$base-class}__img {
      width: 48px;
      height:48px;
      margin-right: 16px;
    }
    
    .#{$base-class}__wrapper {
      display: flex;
      flex-direction: column;
      max-width: 520px;
    }
    
    .#{$base-class}__header {
      font-weight: 600;
    }
    
    .#{$base-class}__footer {
      display: flex;
      flex-shrink: 0;
      margin-top: 12px;
    }
    
    .#{$base-class}__button-text {
      margin-right: 24px;
    }
  }

  &--large {
    min-width: 600px;

    .#{$base-class}__content {
      margin-bottom: 12px;
      display: flex;
      justify-content: flex-start;
      flex-direction: row;
    }

    .#{$base-class}__img {
      width: 48px;
      height:48px;
      margin-right: 20px;
    }
    .#{$base-class}__wrapper {
      display: flex;
      flex-direction: column;
      max-width: 520px;
      min-width: 180px;
    }

    .#{$base-class}__header {
      font-weight: 600;
    }

    .#{$base-class}__footer {
      margin: 0 56px 0 56px;
      display: inline-flex;
      align-items: center;
      flex-shrink: 0;
    }

    .#{$base-class}__button-text {
      margin-right: 24px;
    }
  }

  &__close-icon {
    height: 24px;
    width: 24px;
    cursor: pointer;
    position: absolute;
    top: 17px;
    right: 20px;
  }
}
