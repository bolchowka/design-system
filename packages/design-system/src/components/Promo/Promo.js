import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { Button } from '../Button';

import getMergedClassNames from '../../utils/getMergedClassNames';
import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';

import styles from './style.scss';

const cx = classNames.bind(styles);

const baseClass = 'promo';

export const Promo = props => {
  const {
    buttonText,
    children,
    className,
    header,
    image,
    light,
    linkText,
    onButtonClick,
    onClose,
    onLinkClick,
    size,
    img,
    ...restProps
  } = props;

  const mergedWrapperClassNames = getMergedClassNames(
    cx({
      [`${baseClass}--light`]: light,
      [`${baseClass}--${size}`]: size,
      [`${baseClass}`]: true,
    }),
    className
  );
  const shouldRenderFooter = buttonText || linkText;

  return (
    <div className={mergedWrapperClassNames} {...restProps}>
      <div className={styles[`${baseClass}__content`]}>
        <img src={img} className={styles[`${baseClass}__content-img`]} />
        <div className={styles[`${baseClass}__content-wrapper`]}>
          <div className={styles[`${baseClass}__content-header`]}>{header}</div>
          <div>{children}</div>
        </div>
      </div>
      {shouldRenderFooter && (
        <div className={styles[`${baseClass}__content-footer`]}>
            {buttonText && <Button primary onClick={onButtonClick} className={styles[`${baseClass}__content-footer-button`]}>{buttonText}</Button>}
            {linkText && <Button primary onClick={onLinkClick}>{linkText}</Button>}
        </div>)
      }
      <CloseIcon onClick={onClose} fill="#424d57" className={styles[`${baseClass}__close-icon`]}/>
    </div>
  );
}

Promo.propTypes = {
  buttonText: PropTypes.string,
  children: PropTypes.node.isRequired,
  header: PropTypes.string.isRequired,
  image: PropTypes.node,
  img: PropTypes.string,
  light: PropTypes.bool,
  linkText: PropTypes.string,
  onButtonClick: PropTypes.func,
  onClose: PropTypes.func,
  onLinkClick: PropTypes.func,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
};

Promo.defaultProps = {
  size: 'small',
  img: "https://via.placeholder.com/100"
};

export default Promo;
