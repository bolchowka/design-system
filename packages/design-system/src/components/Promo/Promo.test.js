import * as React from 'react';
import { shallow } from 'enzyme';

import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';

import {Button} from '../Button';
import Promo from './Promo';

describe('Promo', () => {
  const buttonText = "Primary button";
  const header = "This example headline has 40 characters";
  const img = "img";
  const linkText = "Link";
  const onButtonClick = jest.fn();
  const onClose = jest.fn();
  const onLinkClick = jest.fn();
  const props = {
    onClose,
    buttonText,
    header,
    img,
    linkText,
    onButtonClick,
    onClose,
    onLinkClick,
  }
  it('should render promo of any size', () => {
    const sizes = ['small', 'large', 'medium'];
      sizes.map(size => {
        const component = shallow(<Promo size={size} {...props}>Example text</Promo>);

        expect(component).toMatchSnapshot();
      })
  });

  it('should render promo of any size without footer', () => {
    const sizes = ['small', 'large', 'medium'];
    const props = {
      onClose,
      header,
      img,
      onClose,
    }
      sizes.map(size => {
        const component = shallow(<Promo size={size} {...props}>Example text</Promo>);

        expect(component).toMatchSnapshot();
      })
  });

  it('should call on close on close button click', () => {
    const component = shallow(<Promo {...props}>Example text</Promo>);
    component.find('button').simulate('click')
    
    expect(onClose).toHaveBeenCalledTimes(1)
  })

  it('should call buttons actions on buttons click', () => {
    const component = shallow(<Promo {...props}>Example text</Promo>);
    component.find(Button).first().simulate('click')
    component.find(Button).last().simulate('click')
    
    expect(onButtonClick).toHaveBeenCalledTimes(1)
    expect(onLinkClick).toHaveBeenCalledTimes(1)
  })
});
