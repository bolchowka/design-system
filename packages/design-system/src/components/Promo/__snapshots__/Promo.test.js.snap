// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo should render promo of any size 1`] = `
<div
  className="promo--small promo"
>
  <div
    className="promo__content"
  >
    <img
      className="promo__content-img"
      src="img"
    />
    <div
      className="promo__content-wrapper"
    >
      <div
        className="promo__content-header"
      >
        This example headline has 40 characters
      </div>
      <div>
        Example text
      </div>
    </div>
  </div>
  <div
    className="promo__content-footer"
  >
    <ForwardRef
      className="promo__content-footer-button"
      onClick={[MockFunction]}
      primary={true}
    >
      Primary button
    </ForwardRef>
    <ForwardRef
      onClick={[MockFunction]}
      primary={true}
    >
      Link
    </ForwardRef>
  </div>
  <CloseIcon
    className="promo__close-icon"
    fill="#424d57"
    onClick={[MockFunction]}
  />
</div>
`;

exports[`Promo should render promo of any size 2`] = `
<div
  className="promo--large promo"
>
  <div
    className="promo__content"
  >
    <img
      className="promo__content-img"
      src="img"
    />
    <div
      className="promo__content-wrapper"
    >
      <div
        className="promo__content-header"
      >
        This example headline has 40 characters
      </div>
      <div>
        Example text
      </div>
    </div>
  </div>
  <div
    className="promo__content-footer"
  >
    <ForwardRef
      className="promo__content-footer-button"
      onClick={[MockFunction]}
      primary={true}
    >
      Primary button
    </ForwardRef>
    <ForwardRef
      onClick={[MockFunction]}
      primary={true}
    >
      Link
    </ForwardRef>
  </div>
  <CloseIcon
    className="promo__close-icon"
    fill="#424d57"
    onClick={[MockFunction]}
  />
</div>
`;

exports[`Promo should render promo of any size 3`] = `
<div
  className="promo--medium promo"
>
  <div
    className="promo__content"
  >
    <img
      className="promo__content-img"
      src="img"
    />
    <div
      className="promo__content-wrapper"
    >
      <div
        className="promo__content-header"
      >
        This example headline has 40 characters
      </div>
      <div>
        Example text
      </div>
    </div>
  </div>
  <div
    className="promo__content-footer"
  >
    <ForwardRef
      className="promo__content-footer-button"
      onClick={[MockFunction]}
      primary={true}
    >
      Primary button
    </ForwardRef>
    <ForwardRef
      onClick={[MockFunction]}
      primary={true}
    >
      Link
    </ForwardRef>
  </div>
  <CloseIcon
    className="promo__close-icon"
    fill="#424d57"
    onClick={[MockFunction]}
  />
</div>
`;
